frontend/src/shared/helpers.tsx

```diff
@@ -1044,10 +1044,10 @@ export function getSamplePopupParamId(
   samples: SamplesListQuery["samples"],
   paramId: string
 ) {
-  if (parentWhereVariables.OR?.[0].patientsHasSampleConnection_SOME) {
-    const { cmoPatientId, dmpPatientId } = samples[0];
+  if (parentWhereVariables.OR?.[0].patientsHasSample_SOME) {
+    const { cmoPatientId, dmpPatientId } = samples[0] || {};
     if (cmoPatientId) return cmoPatientId;
-    return dmpPatientId;
+    if (dmpPatientId) return dmpPatientId;
```

Comment:
Safer handling of cases where `samples[0]` is undefined or `dmpPatientId` doesn't exist.